src/Jackett.Common/Indexers/Corsarored.cs

```diff
@@ -1,256 +1,247 @@
+		{
+			var result = await RequestStringWithCookiesAndRetry(ApiLatest, null, SiteLink, _apiHeaders);
+			return CheckResponse(result);
+		}
+
+		private string GetApiCategory(TorznabQuery query)
+		{
+			var cats = MapTorznabCapsToTrackers(query);
+			var apiList = new List<string>();
+			foreach (var apiCat in cats.Select(cat =>
+				_apiCategories.FirstOrDefault(kvp => kvp.Value.Contains(cat)).Key))
+			{
+				if (apiCat == "0") return apiCat;
+				if (!apiList.Contains(apiCat))
+					apiList.Add(apiCat);
+				if (apiList.Count > 1) return "0";
+			}
+
+			return apiList[0];
+		}
+
+		protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+		{
+			var releases = new List<ReleaseInfo>();
+			var searchString = query.GetQueryString();
+
+			if (string.IsNullOrWhiteSpace(searchString))
+			{
+				// no term execute latest search
+				var result = await SendApiRequestLatest();
+
+				try
+				{
+					// this time is a jarray
+					var json = (JArray) result;
+
+					releases.AddRange(json.Select(MakeRelease));
+				}
+				catch (Exception ex)
+				{
+					OnParseError(result.ToString(), ex);
+				}
+
+				return releases;
+			}
+
+			var queryCollection = new Dictionary<string, string>
+			{
+				["term"] = searchString,
+				["category"] = GetApiCategory(query)
+			};
+
+			for (var page = 1; page <= MaxSearchPageLimit; page++)
+			{
+				// update page number
+				queryCollection["page"] = page.ToString();
+
+				var result = await SendApiRequest(queryCollection);
+				try
+				{
+					// this time is a jobject
+					var json = (JObject) result;
+
+					// check JSON response and exit if no results (!x.Any())
+					// throws exception if json["results"] is null or not a JArray
+					if (!(json["results"] as JArray ?? throw new Exception("Error invalid JSON response")).Any())
+						break;
+
+					releases.AddRange(json["results"].Select(MakeRelease));
+				}
+				catch (Exception ex)
+				{
+					OnParseError(result.ToString(), ex);
+				}
+			}
+
+			return releases;
+		}
+
+		private ReleaseInfo MakeRelease(JToken torrent)
+		{
+			//https://corsaro.red/details/E5BB62E2E58C654F4450325046723A3F013CD7A4
+			var release = new ReleaseInfo
+			{
+				Title = (string) torrent["title"],
+				Grabs = (long) torrent["completed"],
+				Description = $"{(string) torrent["category"]} {(string) torrent["description"]}",
+				Seeders = (int) torrent["seeders"],
+				InfoHash = (string) torrent["hash"],
+				MagnetUri = new Uri((string) torrent["magnet"]),
+				Comments = new Uri($"{SiteLink}details/{(string) torrent["hash"]}"),
+				DownloadVolumeFactor = 0,
+				UploadVolumeFactor = 1
+			};
+
+			release.Guid = release.Comments;
+			release.Peers = release.Seeders + (int) torrent["leechers"];
+
+			release.PublishDate = DateTime.Now;
+			if (torrent["last_updated"] != null)
+				release.PublishDate = DateTime.Parse((string) torrent["last_updated"]);
+
+			// TODO: don't know how to map this cats..
```

Comment:
if cats are working, remove this todo

Reply:
Todo was from previous author. Will remove